src/shims/fs.rs

```diff
@@ -23,12 +23,23 @@ pub struct FileHandler {
     low: i32,
 }
 
+impl FileHandler {
+    fn next_fd(&self) -> i32 {
+        self.low + 1
+    }
+
+    fn register_fd(&mut self, fd: i32, handle: FileHandle) {
+        self.low = fd;
+        self.handles.insert(fd, handle).unwrap_none();
+    }
+}
+
 impl Default for FileHandler {
     fn default() -> Self {
         FileHandler {
             handles: Default::default(),
             // 0, 1 and 2 are reserved for stdin, stdout and stderr.
-            low: 3,
+            low: 2,
```

Comment:
So looks like you are changing "low"'s meaning from "first fresh FD" to "last used FD"? Please document that in  `struct FileHandler`.

Reply:
The meaning of low isn't changing, as both before and after this PR, low gets preincremented, and the incremented value is used for the new file handle. The change here in the initial value of low just means that fd 3 won't get skipped. (See also the top message in this PR)

Reply:
Ah I see. I don't have a strong preference for it being pre- or post-incremented, just that the variable name makes sense and that things are documented.

In stacked-borrows we stored the next free ID, i.e., we post-increment. So I have a weak preference for that, to keep things consistent across Miri.